### ISSUE TITLE
[Top Bar] Menu Icon (span v button)

### ISSUE THREAD

**Issue by andycochran** [zurb/foundation-sites#8204]
The docs recommend marking up a hamburger menu with… 
```html
<span data-responsive-toggle="responsive-menu" data-hide-for="medium">
  <span class="menu-icon dark" data-toggle></span>
</span>
```

Semantically, shouldn't this be a `<button>` element? 

Downside to using a `<button>`, it gets an outline on click focus. 

http://codepen.io/andycochran/pen/EPzdvG

**Comment by Lynda333** [comment]
They do use `<button>` in the [Responsive Toggle ](http://foundation.zurb.com/sites/docs/responsive-navigation.html#responsive-toggle) section. In the [Top Bar](http://foundation.zurb.com/sites/docs/top-bar.html#advanced) section, it's as you have it. 

Also, the outline doesn't happen in Firefox but does in Chrome.

**Comment by gakimball** [comment]
Makes sense, all of our `.menu-icon` examples should use `<button>`.

You can add the `disable-mouse-outline()` mixin to the CSS to get rid of the outline for mouse users.

**Comment by gakimball** [comment]
@andycochran Are any changes required in the CSS, or just in the examples? And if you want to implement it we'll bring it in :)

**Comment by andycochran** [comment]
@gakimball Should I target 6.2-dev?

**Comment by andycochran** [comment]
@gakimball I think we need to add the `disable-mouse-outline` mixin to the menu's `button` element. Generic `button` elements don't get the What-Input treatment unless they also have the `.button` class. Would it be bad to add What-Input to all `button` elements globally? I can't think of any issues with that, but it's a hefty change I thought might need discussion.

**Comment by gakimball** [comment]
@andycochran Shouldn't be a hefty change, and because `<button>`s are interactive elements it makes sense.

**Comment by gakimball** [comment]
@andycochran If you want to implement this you can target `develop`.

**Comment by gakimball** [comment]
Merged #8370
